crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -1160,4 +1160,28 @@ mod tests {
         world.flush();
         assert_eq!(vec!["event", "event"], world.resource::<Order>().0);
     }
+
+    #[test]
+    fn observer_on_remove_during_despawn_spawn_empty() {
+        let mut world = World::new();
+
+        // Observe the removal of A - this will run during despawn
+        world.observe(|_: Trigger<OnRemove, A>, mut cmd: Commands| {
+            // Spawn a new entity - this reserves a new ID and requires a flush
+            // afterward before Entities::free can be called.
+            cmd.spawn_empty();
+        });
+
+        world.flush();
```

Comment:
Comments for why each of these flushes are needed please.

Reply:
Weren't actually necessary to begin with!